synapse/storage/util/id_generators.py

```diff
@@ -157,12 +188,10 @@ def __init__(
         # The key and values are the same, but we never look at the values.
         self._unfinished_ids: OrderedDict[int, int] = OrderedDict()
 
+    def advance(self, instance_name: str, new_id: int) -> None:
+        raise Exception("Replication is not supported by StreamIdGenerator")
+
```

Comment:
Make `StreamIdGenerator` implement `AbstractStreamIdTracker`. Note that this is no worse than what would currently happen if `advance` were to be called on `StreamIdGenerator`.